test/e2e/apimachinery/aggregator.go

```diff
@@ -570,3 +578,53 @@ func generateFlunderName(base string) string {
 	}
 	return fmt.Sprintf("%s-%d", base, id)
 }
+
+func deleteAPIService(f *framework.Framework, aggrclient *aggregatorclient.Clientset) {
+	// This function allows for clean execution when having HA control planes. We need to allow time for
+	// the sample api server to be unregistered from all api servers before tearing down the deployment.
+	framework.Logf("Waiting on v1alpha1.wardle.example.com to disappear from all apiserver instances")
+	client := f.ClientSet
+
+	err := aggrclient.ApiregistrationV1().APIServices().Delete(context.TODO(), "v1alpha1.wardle.example.com", metav1.DeleteOptions{})
+	framework.ExpectNoError(err, "deleting apiservice %s", "v1alpha1.wardle.example.com")
+
+	endpoints, err := client.CoreV1().Endpoints("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
```

Comment:
Not sure why I can't answer the first comment. Adding here.

>everything in the test namespace gets deleted automatically and will not conflict between tests
cluster-scoped things and things in kube-system need explicit cleanup

Even between tests within the same `Describe`? Will remove the useless `defer` statements then.

Reply:
> I think running an exec in a pod to poll discovery is more likely to be fragile and fail than simply waiting a few seconds (5?) to give HA API servers time to observe the APIService delete.

Problem about setting a timeout is that its a best guess, so it might be ok today and short tomorrow, or fragile under different conditions. Also, it might introduce flakes, which are usually quite confusing.
The alternative discussed above would be to introduce this functionality in `agnhost`, but thats quite a lot of boilerplate and side tasks to execute just one command in a well known pattern, as many tests make use of exec pods and curl.

Reply:
> Problem about setting a timeout is that its a best guess, so it might be ok today and short tomorrow, or fragile under different conditions. Also, it might introduce flakes, which are usually quite confusing.

Running a pod and execing into it to check a condition also introduces more possible failure modes and potential for flakes

Reply:
> The alternative discussed above would be to introduce this functionality in agnhost, but thats quite a lot of boilerplate and side tasks to execute just one command in a well known pattern, as many tests make use of exec pods and curl.

That would still require running a pod to verify an operation that should complete in *well* under a second.

Reply:
> I guess I would claim that if members of an HA cluster are _not_ reacting to removal of an APIService within a ~second, that's a problem for the system that will have negative impacts on users of that system,

the curl in pod thing has demonstrated to be reliable, most of the sig-network pods use it, but I agree with this statement, the key is to define that threshold for _time to propagate a change between apiservers_